EduNLP/SIF/segment/segment.py

```diff
@@ -116,7 +119,11 @@ def __len__(self):
 
     def append(self, segment) -> None:
         if isinstance(segment, TextSegment):
-            self._text_segments.append(len(self))
+            if len(self._text_segments) != 0 and self._text_segments[-1] == len(self) - 1:
```

Comment:
Why modify these lines?

Reply:
I think textf{} can be simply seen as a TextSegment

Reply:
As you said,
> "hello world, I am \textf{Robot, b}" should be aggregate into one text_segments as ['hello world, I am Robot']
Without the if branch, it will be divided into ['hello world, I am', 'Robot']

Reply:
However, why not first judge whether the text conatins textf and process it with regex?

Reply:
I do not think it is a good way to process it in two steps for you break the original code logic withour enough test cases which in fact has already failed in the test stage.

Reply:
Got it, let me give it another shot